src/executor_config/barrier.md

```diff
@@ -0,0 +1,89 @@
+-TODO: All Barriers Explained
+-TODO: Security Note about misconfigured barriers
+-TODO: Example
+
+# Barrier
+Before any XCMs are executed in the XCM executor, they must pass the Barrier.
+The Barrier type implements the `ShouldExecute`
+[trait](https://paritytech.github.io/polkadot/doc/xcm_executor/traits/trait.ShouldExecute.html)
+and can be viewed as the firewall for the xcm-executor. 
+Each time the xcm-executor receives an XCM, it checks with the barrier to determine if the XCM should be executed.
+
+We can also define multiple barriers for our Barrier type using a tuple.
+During execution, each barrier is checked, and if any of them succeed, the XCM is executed.
```

Comment:
question for me: how can you express an `&&` operator with these barriers then? is there a way to say: Execute if _all_ of my barriers pass?

Reply:
As far as I know we do not support `&&` operator in any way as I am not sure if it makes sense with the current barrier implementations. Most of these implementations are used for specific XCMs. We of course already have nested barriers like the WithComputedOrigin. We could make something similar that takes other implementations and only passes if all the implementations pass.